notebooks/proj_sys_eng/sitcom_integration/l3_system_integ/LVV-T2216-plots.ipynb

```diff
@@ -0,0 +1,1166 @@
+    "### Analyse M2Hex"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7daa848d-3c15-49a4-8fd1-0cd6253a3e30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From the XML:\n",
+    "#   Actual MTHexapod position, in order (X, Y, Z, U, V, W). \n",
+    "#   Linear positions are in microns, angular positions are in degrees.\n",
+    "pos = await client.select_time_series(\n",
+    "    \"lsst.sal.MTHexapod.application\",\n",
+    "    \"*\",\n",
+    "    t_start,\n",
+    "    t_end,\n",
+    "    index=2\n",
+    ")\n",
+    "\n",
+    "# Unravel in x/y/z/u/v/w\n",
+    "for i, col in enumerate(\"xyzuvw\"):\n",
+    "    pos[col] = pos[f\"position{i}\"]\n",
+    "\n",
+    "\n",
+    "# Triggered at the end of a slew\n",
+    "cpos = await client.select_time_series(\n",
+    "    \"lsst.sal.MTHexapod.logevent_compensatedPosition\",\n",
+    "    \"*\",\n",
+    "    t_start,\n",
+    "    t_end,\n",
+    "    index=2\n",
+    ")\n",
+    "\n",
+    "# Triggered only after move/offset. Should not see much. \n",
+    "upos = await client.select_time_series(\n",
+    "    \"lsst.sal.MTHexapod.logevent_uncompensatedPosition\",\n",
+    "    \"*\",\n",
+    "    t_start,\n",
+    "    t_end,\n",
+    "    index=2\n",
+    ")\n",
+    "\n",
+    "# Estimate the LUT position\n",
+    "lut_pred = vandv.hexapod.get_lut_positions(index=2, elevation=el.actualPosition)\n",
+    "lut = pd.DataFrame(lut_pred, columns=[\"x\", \"y\", \"z\", \"u\", \"v\", \"w\"], index=el.index)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b67bb1d-e937-49b4-81b7-0cda0ade2689",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axs = plt.subplot_mosaic(\n",
+    "    mosaic=\"AD\\nBE\\nCF\",\n",
+    "    num=\"Slew Without LUT Correction - M2Hex Actual Position\", \n",
+    "    dpi=120,\n",
+    "    figsize=(10, 6),\n",
+    "    sharex=True,\n",
+    ")\n",
+    "\n",
+    "cols = \"xyzuvw\"\n",
+    "for ax, col in zip(\"ABCDEF\", cols):\n",
+    "    _ = vandv.hexapod.timeline_position(\n",
+    "        axs[ax], \n",
+    "        [pos, cpos, upos, lut], \n",
+    "        column=col, \n",
+    "        elevation=el, \n",
+    "        symbols=[\"\", \"o\", \"s\", \"\"],\n",
+    "        names=[\"Actual Position\", \"Compensated\", \"Uncompensated\", \"LUT\"]\n",
+    "    )\n",
+    "\n",
+    "# Hide xlabel\n",
+    "for i in \"ABDE\":\n",
+    "    _ = axs[i].set_xlabel(\"\")\n",
+    "    \n",
+    "_ = axs[\"F\"].legend(loc='lower center', bbox_to_anchor=(0.5, -1), ncol=5)\n",
+    "\n",
+    "fig.suptitle(\"M2Hex Timeline - Slew w/ Compensation\")\n",
+    "fig.autofmt_xdate()\n",
+    "fig.tight_layout()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b51c6dd-ad03-4f15-a871-28409becd258",
+   "metadata": {},
+   "source": [
+    "## Compare Snapshots with LUT\n",
+    "\n",
+    "Compare the average snapshots in different elevations using the LUT."
```

Comment:
This snapshot comparison will be so useful, as for instance in LVV-T2214. :)

Reply:
Again, the snapshots receive a series. You can use Pandas functionality to select the row that will be shown in the snapshot.